internal/pkg/opa/default-compiler.go

```diff
@@ -93,84 +92,94 @@ func (compiler *policyCompiler) Configure(appConf *configs.AppConfig, compConf *
 		}
 	}
 
 	rawInput, exists := requestBody["input"]
 	if !exists {
-		compiler.handleError(w, uid, internalErrors.InvalidInput{Msg: "PolicyCompiler: Incoming request had no field 'input'!"})
+		compiler.handleError(w, internalErrors.InvalidInput{Msg: "PolicyCompiler: Incoming request had no field 'input'!"})
 		return
 	}
 	input, ok := rawInput.(map[string]interface{})
 	if !ok {
-		compiler.handleError(w, uid, internalErrors.InvalidInput{Msg: "PolicyCompiler: Field 'input' in request body was no nested JSON object!"})
+		compiler.handleError(w, internalErrors.InvalidInput{Msg: "PolicyCompiler: Field 'input' in request body was no nested JSON object!"})
 		return
 	}
-	logging.LogForComponent("policyCompiler").WithField("UID", uid).Debugf("Received input: %+v", input)
+	logging.LogForComponent("policyCompiler").Debugf("Received input: %+v", input)
 
 	// Process path
 	output, err := compiler.processPath(input)
 	if err != nil {
-		compiler.handleError(w, uid, err)
+		compiler.handleError(w, err)
+		return
+	}
+
+	path, errPath := extractURLFromRequestBody(input)
+	method, errMethod := input["method"].(string)
+	if errPath != nil || !errMethod {
+		compiler.handleError(w, err)
 		return
 	}
 
 	// Compile mapped path
 	queries, err := compiler.opaCompile(req, input, output)
 	if err != nil {
-		compiler.handleError(w, uid, errors.Wrap(err, "PolicyCompiler: Error during policy compilation"))
+		compiler.handleError(w, errors.Wrap(err, "PolicyCompiler: Error during policy compilation"))
 		return
 	}
 
 	// OPA decided denied
 	if queries.Queries == nil {
-		compiler.writeDeny(w, uid)
+		compiler.writeDeny(w)
+		logging.LogAccessDecision(compiler.config.AccessDecisionLogLevel, path.String(), method, time.Since(startTime).String(), "DENY", "policyCompiler")
 		return
 	}
 	// Check if any query succeeded
 	if done := anyQuerySucceeded(queries); done {
-		compiler.writeAllow(w, uid)
+		compiler.writeAllow(w)
+		logging.LogAccessDecision(compiler.config.AccessDecisionLogLevel, path.String(), method, time.Since(startTime).String(), "ALLOW", "policyCompiler")
 		return
 	}
 
 	// Otherwise translate ast
 	result, err := (*compiler.config.Translator).Process(queries, output.Datastore, req)
 	if err != nil {
-		compiler.handleError(w, uid, errors.Wrap(err, "PolicyCompiler: Error during ast translation"))
+		compiler.handleError(w, errors.Wrap(err, "PolicyCompiler: Error during ast translation"))
 		return
 	}
 
 	// If we receive something from the datastore, the query was successful
 	if result {
-		compiler.writeAllow(w, uid)
+		compiler.writeAllow(w)
+		logging.LogAccessDecision(compiler.config.AccessDecisionLogLevel, path.String(), method, time.Since(startTime).String(), "ALLOW", "policyCompiler")
 	} else {
-		compiler.writeDeny(w, uid)
+		compiler.writeDeny(w)
+		logging.LogAccessDecision(compiler.config.AccessDecisionLogLevel, path.String(), method, time.Since(startTime).String(), "DENY", "policyCompiler")
```

Comment:
I think we should really think again about moving the logging logic into the writeDeny/Allow methods. Because changing three lines when changing something in the logic is a bit messy. You could encapsulate the arguments that are needed for logging in a separate struct 'decisionContext' or so to keep the API of these methods cleaner.